extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -496,6 +501,18 @@ export class AmpStoryPage extends AMP.BaseElement {
     if (this.animationManager_) {
       this.animationManager_.cancelAll();
     }
+
+    this.story360components_.forEach((componentPromise) => {
+      componentPromise.then((component) => {
+        component
+          .signals()
+          .whenSignal(CommonSignals.LOAD_END)
+          .then(() => {
```

Comment:
It'd be great if this would behave like a `HTMLMediaElement` (video or audio). We don't need to wait for it to load to call playback methods.
Can we somehow register the `play/pause/rewind()` from the `amp-story-360` class, and perform the right playback operation on load? It'd make the code here much easier.

Reply:
Okay. That will require some promise queue pattern. It might make the code here easier, but also more complex overall. I wish you'd told me that a week ago 😭

Reply:
Are you sure you need a promise queue? Could you do something really simple in amp-story-360, where when the element is loaded you check if the `this.isPlaying_` status to maybe call play?
If you add some kind of `this.isReady_` boolean to not call `this.animate_()` on public `play()` it looks like it could work?

Reply:
Hmm we'd need to take care of race conditions if eg a pause is requested in the meantime. So basically we'd end up with a promise pattern even if we don't use promises, don't you think?

Reply:
I thought it would be trivial if we did something like the pseudo code below. You're way more familiar with the implementation, if it actually is way more complicated than I thought feel free to disregard my comment.

```
layoutCallback()
  whenLoaded().then(() => {
    if (isPlaying)
      animate()
  });

play()
  isPlaying = true

pause()
  isPlaying = false
```

As a side note, I think this would be much safer. With your current implementation, the LOAD_END promise could resolve AFTER a user left the page, which would still start playing the animation.
This issue should be resolved by the implementation I'm suggesting.

Reply:
Yes you're right, that makes sense and is simple enough.